git/difffiles.go

```diff
@@ -52,15 +52,20 @@ func DiffFiles(c *Client, opt DiffFilesOptions, paths []File) ([]HashDiff, error
 			val = append(val, HashDiff{idx.PathName, idxtree, fs, uint(idx.Fsize), 0})
 			continue
 		}
-		stat, err := f.Stat()
+		stat, err := f.Lstat()
 		if err != nil {
 			val = append(val, HashDiff{idx.PathName, idxtree, fs, uint(idx.Fsize), 0})
 			continue
 		}
 
 		switch {
 		case stat.Mode().IsDir():
-			fs.FileMode = ModeTree
+			// Since we're diffing files in the index (which only holds files)
+			// against a directory, it means that the file was deleted and
+			// replaced by a directory.
+			//fs.FileMode = ModeTree
```

Comment:
Same comment as in the other PR. I'm not a fan of commented out code. The code around can deviate and make it so that it no longer compiles. It affects readability of the code. Git can give us the history so that we can resurrect it if necessary. :)